lib/sycamore/sycamore/tests/unit/query/test_schema.py

```diff
@@ -0,0 +1,67 @@
+from unittest.mock import MagicMock
+
+from sycamore.query.schema import OpenSearchSchemaFetcher
+
+
+def test_opensearch_schema():
+    mock_client = MagicMock()
+    mock_client.get_field_mapping.return_value = {
+        "test_index": {
+            "mappings": {
+                "properties.entity.day": {"full_name": "properties.entity.day", "mapping": {"day": {"type": "date"}}},
+                "properties.entity.aircraft": {
+                    "full_name": "properties.entity.aircraft",
+                    "mapping": {
+                        "aircraft": {"type": "text", "fields": {"keyword": {"type": "keyword", "ignore_above": 256}}}
+                    },
+                },
+                "properties.entity.location": {
+                    "full_name": "properties.entity.location",
+                    "mapping": {
+                        "location": {"type": "text", "fields": {"keyword": {"type": "keyword", "ignore_above": 256}}}
+                    },
+                },
+            }
+        }
+    }
+
+    mock_query_executor = MagicMock()
+    # Note that there are no values for 'location' here.
+    mock_query_executor.query.return_value = {
+        "result": {
+            "hits": {
+                "hits": [
+                    {
+                        "_source": {
+                            "properties": {
+                                "entity": {
```

Comment:
maybe have an entry that is missing a property

Reply:
Done.